server/sonar-server/src/main/java/org/sonar/server/email/ws/UpdateAction.java

```diff
@@ -0,0 +1,139 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.email.ws;
+
+import javax.annotation.Nullable;
+import org.sonar.api.config.Settings;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.property.PropertyDto;
+import org.sonar.server.user.UserSession;
+
+import static org.sonar.api.config.EmailSettings.FROM;
+import static org.sonar.api.config.EmailSettings.FROM_DEFAULT;
+import static org.sonar.api.config.EmailSettings.PREFIX;
+import static org.sonar.api.config.EmailSettings.PREFIX_DEFAULT;
+import static org.sonar.api.config.EmailSettings.SMTP_HOST;
+import static org.sonar.api.config.EmailSettings.SMTP_PASSWORD;
+import static org.sonar.api.config.EmailSettings.SMTP_PORT;
+import static org.sonar.api.config.EmailSettings.SMTP_PORT_DEFAULT;
+import static org.sonar.api.config.EmailSettings.SMTP_SECURE_CONNECTION;
+import static org.sonar.api.config.EmailSettings.SMTP_USERNAME;
+
+public class UpdateAction implements EmailsWsAction {
+
+  private static final String PARAM_HOST = "host";
+  private static final String PARAM_PORT = "port";
+  private static final String PARAM_SECURE = "secure";
+  private static final String PARAM_USERNAME = "username";
+  private static final String PARAM_PASSWORD = "password";
+  private static final String PARAM_FROM = "from";
+  private static final String PARAM_PREFIX = "prefix";
+
+  private static final String SSL_VALUE = "ssl";
+  private static final String STARTTLS_VALUE = "starttls";
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+
+  // TODO remove this when EmailSettings won't use Settings in memory to get settings
+  private final Settings settings;
+
+  public UpdateAction(DbClient dbClient, UserSession userSession, Settings settings) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.settings = settings;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction("update")
+      .setDescription("Update email settings<br>" +
+        "Requires 'Administer System' permission")
+      .setSince("6.1")
+      .setInternal(true)
+      .setPost(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_HOST)
+      .setDescription("SMTP host. Leave blank to disable email sending.")
```

Comment:
after discussion: all parameters are optional and it fails if an empty value is provided